src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java

```diff
@@ -307,16 +309,32 @@ public boolean teleport(@NotNull Location location)
 	}
 
 	@Override
+	@SuppressWarnings("UnstableApiUsage")
 	public boolean teleport(@NotNull Location location, @NotNull TeleportCause cause)
 	{
-		return teleport(location, cause, false);
+		return teleport(location, cause, new TeleportFlag[0]);
 	}
 
 	@Override
-	public boolean teleport(@NotNull Location location, @NotNull TeleportCause cause, boolean ignorePassengers, boolean dismount)
+	@SuppressWarnings("UnstableApiUsage")
+	public boolean teleport(@NotNull Location location, @NotNull TeleportCause cause, TeleportFlag @NotNull ... flags)
 	{
 		Preconditions.checkNotNull(location, "Location cannot be null"); // The world can be null if it's not a player
 		location.checkFinite();
+
+		Set<TeleportFlag> flagSet = Set.of(flags);
+		boolean dismount = !flagSet.contains(TeleportFlag.EntityState.RETAIN_VEHICLE);
+		boolean ignorePassengers = flagSet.contains(TeleportFlag.EntityState.RETAIN_PASSENGERS);
+
+		if (flagSet.contains(TeleportFlag.EntityState.RETAIN_PASSENGERS) && this.hasPassengers() && location.getWorld() != this.getWorld())
```

Comment:
`location.getWorld() != this.getWorld()`

Is this the right way to compare these?

Reply:
Yes. `#equals` isn't overridden in MockBukkit or Paper.